### PR TITLE
New protocol

### PR DESCRIPTION
**Pull request by Tocutoeltuco**
This branch uses the new transformice protocol, which changes the packet size format.
It is required to update to this new version in order to keep using transfromage.

### CODE REVIEW

**Comment by Tocutoeltuco** [comment]
By the way, I didn't test it. It might have some bugs, but I guess it does work.

File: libs/client.lua [hunk 1]
```diff
@@ -1232,7 +1232,8 @@ packetListener = {
 	},
 	[44] = {
 		[1] = function(self, packet, connection, identifiers) -- Switch bulle identifiers
-			local bulleId = packet:read32()
+			local bulleCred1 = packet:read32()
+			local bulleCred2 = packet:read32()
```
Comment (Lautenschlager-id) [inline libs/client.lua]: I don't really like the name of these variables...

File: libs/connection.lua [hunk 2]
```diff
@@ -46,7 +49,10 @@ connection.new = function(self, name, event)
 		packetID = 0,
 		port = 1,
 		name = name,
-		open = false
+		open = false,
+		length_bytes = 0,
```
Comment (Lautenschlager-id) [inline libs/connection.lua]: Lua uses camelCase

File: libs/connection.lua [hunk 3]
```diff
@@ -46,7 +49,10 @@ connection.new = function(self, name, event)
...
+		length = 0,
+		read_length = false
```
Comment (Lautenschlager-id) [inline libs/connection.lua]: Lua uses camelCase

File: libs/connection.lua [hunk 4]
```diff
@@ -155,18 +170,16 @@ connection.send = function(self, identifiers, alphaPacket)
...
-		gammaPacket = byteArray:new():write8(3):write24(stackLen)
-	else
-		return error("↑failure↓[SEND]↑ The packet length is too big! ↑error↓(" .. stackLen .. ")↑\n\tIdentifiers: " .. table_concat(identifiers, ','), enum.errorLevel.low)
+	local gammaPacket = byteArray:new()
+	local size = #betaPacket.stack
+	local size_type = bit_rshift(size, 7)
+	while size_type ~= 0 do
+		gammaPacket:write8(bit_bor(bit_band(size, 127), 128))
+		size = size_type
+		size_type = bit_rshit(size_type, 7)
```
Comment (Lautenschlager-id) [inline libs/connection.lua]:
```suggestion
		size_type = bit_rshift(size_type, 7)
```

File: libs/connection.lua [hunk 5]
```diff
@@ -114,20 +120,29 @@ end
...
-			stackLen = stackLen + bit_lshift(byteArr[i], (8 * (stackLenSize - i)))
-		end
+		self.length_bytes = 0
+		self.length = 0
+		self.read_length = false
 
-		return self.buffer:receive(stackLen)
+		return byteArr
+	else
+		return
```
[the remark this answers is not in the document]
Comment (Tocutoeltuco) [reply libs/connection.lua]: No other part of the code rely on it. It's useless.

File: libs/client.lua (hunk 1 above)
Comment (Tocutoeltuco) [reply libs/client.lua]: How would you name it then?

**Comment by Tocutoeltuco** [comment]
By the way, @Minstfm has tested this PR and he said it runs nicely.

**Comment by Seniru** [comment]
Worked for me too :+1:

**Pull request merged**